tests/trig.rs

```diff
@@ -153,46 +153,44 @@ impl SynthLanguage for Trig {
     fn get_lifting_rules() -> Ruleset<Self> {
         Ruleset::new(&[
             // definition of sine, cosine, tangent
+            // (sine)
             "(sin ?a) ==> (/ (- (cis ?a) (cis (~ ?a))) (* 2 I))",
             "(/ (- (cis ?a) (cis (~ ?a))) (* 2 I)) ==> (sin ?a)",
+            // (sine, alternatively)
+            "(sin ?a) ==> (/ (- (* I (cis (~ ?a))) (* I (cis ?a))) 2)",
+            "(/ (- (* I (cis (~ ?a))) (* I (cis ?a))) 2) => (sin ?a)",
+            // (cosine)
             "(cos ?a) ==> (/ (+ (cis ?a) (cis (~ ?a))) 2)",
             "(/ (+ (cis ?a) (cis (~ ?a))) 2) ==> (cos ?a)",
+            // (cosine, alternatively)
+            "(cos ?a) ==> (/ (+ (* I (cis ?a)) (* I (cis (~ ?a)))) (* 2 I))",
+            "(/ (+ (* I (cis ?a)) (* I (cis (~ ?a)))) (* 2 I)) ==> (cos ?a)",
+            // (tangent, alternatively)
             "(tan ?a) ==> (* I (/ (- (cis (~ ?a)) (cis ?a)) (+ (cis (~ ?a)) (cis ?a))))",
             "(* I (/ (- (cis (~ ?a)) (cis ?a)) (+ (cis (~ ?a)) (cis ?a)))) ==> (tan ?a)",
-            // definition of cosecant, secant, cotangent
-            "(csc ?a) ==> (/ 1 (sin ?a))",
-            "(/ 1 (sin ?a)) ==> (csc ?a)",
-            "(sec ?a) ==> (/ 1 (cos ?a))",
-            "(/ 1 (cos ?a)) ==> (sec ?a)",
-            "(cot ?a) ==> (/ 1 (tan ?a))",
-            "(/ 1 (tan ?a)) ==> (cot ?a)",
             // relating tangent to sine and cosine
             "(tan ?a) ==> (/ (sin ?a) (cos ?a))",
```

Comment:
what happens if you remove tan = sin / cos ? does this affect the result?

Reply:
Apparently not. We don't try very hard for tan rules though, so our exploration is very shallow.

Reply:
Maybe I should move to the prior set?

Reply:
Just kidding, they are needed

Reply:
good to know!